src/transformers/trainer.py

```diff
@@ -2255,17 +2255,21 @@ def _inner_training_loop(
                     # Optimizer step
                     self.optimizer.step()
                     optimizer_was_run = not self.accelerator.optimizer_step_was_skipped
+                    # Store the current LR before stepping
+                    learning_rate = self._get_learning_rate()
```

Comment:
I had to modify the order slightly, as we need to grab the LR *before* scheduling, but we need to `step()` it before calling `on_step_end`